tests/phpunit/unit-tests/class-llms-test-generator.php

```diff
@@ -8,9 +8,86 @@
  * @since 3.36.3 Add tests for `is_generator_valid()` and `set_generator()` methods.
  *              Split `is_error()` method tests into multiple tests.
  * @since 3.37.4 Don't test against core metadata.
+ * @since [version] Add tests for image sideloading methods.
  */
 class LLMS_Test_Generator extends LLMS_UnitTestCase {
 
+	/**
+	 * Test create_reusable_block() when the block already exists
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function test_create_reusable_block_already_exists() {
+
+		$gen = new LLMS_Generator( array() );
+
+		$title   = 'Dupcheck reuse block';
+		$content = 'Block content';
+
+		$dup = $this->factory->post->create( array(
+			'post_type' => 'wp_block',
+			'post_title' => $title,
+			'post_content' => $content,
```

Comment:
```suggestion
			'post_content' => $content,
			'post_status'  => 'publish',
```
otherwise the block will be created as 'draft' and you'll just reference a block that technically "doesn't exist" (that's how gutenberg sees it).